src/java.desktop/share/classes/javax/swing/JPopupMenu.java

```diff
@@ -941,10 +941,29 @@ public Component getInvoker() {
     public void setInvoker(Component invoker) {
         Component oldInvoker = this.invoker;
         this.invoker = invoker;
+        PropertyChangeListener propListener = new PropertyChangeListener() {
+            public void propertyChange(PropertyChangeEvent e) {
+                String propertyName = e.getPropertyName();
+                if (propertyName == "ancestor") {
```

Comment:
Just curious, where is the `ancestor` property added to the listener list? I have tried tracking this down to follow the logic, and I could not find it, nor can I find it under the bound properties list on javadoc.

Testing seems to be ok AFAIK but I wanted to trace the code to double-check.

Reply:
The changes to the `ancestor` property are fired from `JComponent.addNotify` and `removeNotify` correspondingly:

https://github.com/openjdk/jdk/blob/9dd93c6a2c5fb4c3a9f2a063a7ab402f9292ad03/src/java.desktop/share/classes/javax/swing/JComponent.java#L4837

https://github.com/openjdk/jdk/blob/9dd93c6a2c5fb4c3a9f2a063a7ab402f9292ad03/src/java.desktop/share/classes/javax/swing/JComponent.java#L4858

Reply:
Guess  I already mentioned in the description